### PR TITLE
Add Star Compass and refactor Grappling hook to work with new geary API

### PR DESCRIPTION
**Pull request by Derongan**

### CODE REVIEW

File: src/main/java/com/derongan/minecraft/mineinabyss/MineInAbyss.java [hunk 1]
```diff
@@ -49,31 +59,39 @@ public void onEnable() {
...
-                () -> PredefinedArtifacts
-                    .createGrapplingHook(1.3, 3, 4, Color.fromRGB(142, 89, 60), 1, 64), itemStack);
-
-            recipe.shape("III", "ISI", " S ");
-            recipe.setIngredient('I', Material.IRON_INGOT);
-            recipe.setIngredient('S', Material.STRING);
-
-            getServer().addRecipe(recipe);
+            NamespacedKey grapplingRecipeKey = new NamespacedKey(this, "grappling_hook");
+            NamespacedKey starCompassRecipeKey = new NamespacedKey(this, "star_compas");
```
Comment (lizelive) [inline src/main/java/com/derongan/minecraft/mineinabyss/MineInAbyss.java:71]: this should totally be a json file

File: src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import com.derongan.minecraft.mineinabyss.AbyssContext
+import com.derongan.minecraft.mineinabyss.world.Layer
+import com.mineinabyss.geary.ecs.Family
+import com.mineinabyss.geary.ecs.component.components.control.Activated
+import com.mineinabyss.geary.ecs.entity.GearyEntity
+import com.mineinabyss.geary.ecs.system.IteratingSystem
+import org.bukkit.ChatColor
+import org.bukkit.Location
+
+class AbyssLocationSystem(private val context: AbyssContext) : IteratingSystem(Family.builder().setAll(setOf(Activated::class.java, DepthMeter::class.java)).build()) {
```
Comment (lizelive) [inline src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt:13]: should this be in deeper world?

Comment (0ffz) [reply src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt:13]: I think DeeperWorld should only hold a global coordinate system, the abyssal depth seems pretty specific to us. Though once we have a system like that in DeeperWorld, we could probably use it here for layer depth calculations.

File: src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        entity.removeComponent(Activated::class.java)
+    }
+
+    // TODO memoize total depth of each layer
+    private fun getDepth(layer: Layer, section: Section, location: Location): Int {
+        var totalDepth = 0
+        var currentSectionTop = 0
+        val numSections = layer.sections!!.size
+        layer.sections?.forEachIndexed { index, s ->
+            run {
```
Comment (0ffz) [inline src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt:40]: Does the run block do anything here?

File: src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt (hunk 2 above)
Comment (Derongan) [reply src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt:13]: Offz is correct

File: src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt (hunk 3 above)
Comment (Derongan) [reply src/main/java/com/derongan/minecraft/mineinabyss/geary/AbyssLocationSystem.kt:40]: side effect sets currentSectionTop and totalDepth

File: src/main/java/com/derongan/minecraft/mineinabyss/MineInAbyss.java (hunk 1 above)
Comment (Derongan) [reply src/main/java/com/derongan/minecraft/mineinabyss/MineInAbyss.java:71]: Totally should be- I have a bunch of code from before geary was nbt based that allowed defining prefabs/recipes in geary. I'm repurposing it soon tm

**Pull request merged**